ocp_resources/resource.py

```diff
@@ -1238,22 +1248,21 @@ def wait_for_condition(self, condition: str, status: str, timeout: int = 300, sl
         self.logger.info(f"Wait for {self.kind}/{self.name}'s '{condition}' condition to be '{status}'")
 
         timeout_watcher = TimeoutWatch(timeout=timeout)
-        for sample in TimeoutSampler(
-            wait_timeout=timeout,
-            sleep=sleep_time,
-            func=lambda: self.exists,
-        ):
-            if sample:
-                break
-
+        self.wait(timeout=timeout, sleep=sleep_time)
         for sample in TimeoutSampler(
             wait_timeout=timeout_watcher.remaining_time(),
             sleep=sleep_time,
             func=lambda: self.instance,
         ):
             if sample:
                 for cond in sample.get("status", {}).get("conditions", []):
-                    if cond["type"] == condition and cond["status"] == status:
+                    actual_condition = {"type": cond["type"], "status": cond["status"]}
+                    expected_condition = {"type": condition, "status": status}
+                    if reason is not None:
+                        actual_condition["reason"] = cond.get("reason", "")
+                        expected_condition["reason"] = reason
```

Comment:
if you create `expected_condition` before the loop and set `expected_condition["reason"] = reason` then actual_condition["reason"] can default to None